util/multi_concat.py

```diff
@@ -14,6 +14,50 @@
 from xarray.core.alignment import align
 
 
+try:
+    # solely for isinstance checks
+    import dask.array as dask_array
+
+    dask_array_type = (dask_array.Array,)
+except ImportError:  # pragma: no cover
+    dask_array_type = ()
+    dask_array = None
+
+
+def asarray(data):
+    return (
+        data
+        if (isinstance(data, dask_array_type) or hasattr(data, "__array_function__"))
```

Comment:
Is it possible the "import dask.array..." fails? If so does that impact this code?

Reply:
If the dask array import fails, dask_array_type is set to () [i.e. an empty tuple]. In that event, the first piece of the if statement is checking isinstance(data, ()). This will fail for any value of data, including if data is an empty tuple (or a non-empty tuple). Basically, it is calling isinstance with empty type info which should always return False.